app/src/main/java/com/microsoft/office365/connectmicrosoftgraph/ConnectActivity.java

```diff
@@ -113,7 +114,11 @@ public void onError(Exception exc) {
 
         AuthenticationManager mgr = AuthenticationManager.getInstance();
         mgr.setContextActivity(this);
-        mgr.connect(callback);
+        try {
+            mgr.connect(callback);
+        } catch (AuthenticationException ae) {
```

Comment:
Duplicate error handling structures.
ADAL should return the error in the `onError` method of the callback.
File an issue in their repo.